master/internal/api_workspace.go

```diff
@@ -495,6 +763,37 @@ func (a *apiServer) PatchWorkspace(
 			}
 		}
 		insertColumns = append(insertColumns, "checkpoint_storage_config")
+
+		toDelete := []string{}
+		for cluster, metadata := range req.Workspace.ClusterNamespaceMeta {
+			if metadata.Namespace == nil && !metadata.AutoCreateNamespace {
+				toDelete = append(toDelete, cluster)
+				delete(req.Workspace.ClusterNamespaceMeta, cluster)
+			}
+		}
+		if len(toDelete) > 0 {
+			err = a.deleteWorkspaceNamespaceBinding(ctx, toDelete, req.Id, &currUser)
+			if err != nil && !strings.Contains(err.Error(), "tried to delete default binding.") {
```

Comment:
Comparing string contents of error messages is bad practice; is there a way to detect this case through something like `errors.Is`?